libs/product/src/reducers/configurable-product-entities/configurable-product-entities.reducer.ts

```diff
@@ -0,0 +1,83 @@
+import { EntityState } from '@ngrx/entity';
+
+import { DaffProductGridActionTypes, DaffProductGridActions } from '../../actions/product-grid.actions';
+import { DaffProductActionTypes, DaffProductActions } from '../../actions/product.actions';
+import { DaffBestSellersActionTypes, DaffBestSellersActions } from '../../actions/best-sellers.actions';
+import { daffConfigurableProductAppliedAttributesEntitiesAdapter } from './configurable-product-entities-reducer-adapter';
+import { DaffProduct, DaffProductTypeEnum } from '../../models/product';
+import { DaffConfigurableProductActions, DaffConfigurableProductActionTypes } from '../../actions/configurable-product.actions';
+import { DaffProductVariantAttributesDictionary } from '../../models/configurable-product';
+
+/**
+ * Reducer function that catches actions and changes/overwrites product entities state.
+ * 
+ * @param state current State of the redux store
+ * @param action ProductGrid, BestSellers, or Product actions
+ * @returns Product entities state
+ */
+export function daffConfigurableProductEntitiesReducer<T extends DaffProduct>(
+  state = daffConfigurableProductAppliedAttributesEntitiesAdapter().getInitialState(), 
+  action: DaffProductGridActions<T> | DaffBestSellersActions<T> | DaffProductActions<T> | DaffConfigurableProductActions): EntityState<DaffProductVariantAttributesDictionary> {
+	const adapter = daffConfigurableProductAppliedAttributesEntitiesAdapter();
+  switch (action.type) {
+    case DaffProductGridActionTypes.ProductGridLoadSuccessAction:
+			return adapter.upsertMany(buildConfigurableProductAppliedAttributesEntities(action.payload), state);
+    case DaffBestSellersActionTypes.BestSellersLoadSuccessAction:
+      return adapter.upsertMany(buildConfigurableProductAppliedAttributesEntities(action.payload), state);
```

Comment:
it seems that these 2 cases can be combined.

Reply:
done